app/controllers/concerns/blacklight/catalog.rb

```diff
@@ -98,6 +95,24 @@ def facet
     end
   end
 
+  # displays facet values that are relevant both to the current search and a facet_suggestion_query
+  def facet_suggest
+    retrieve_facet_by_id params[:facet_id]
+    query_fragment = params[:query_fragment] || ''
+
+    @response = if query_fragment.present?
+                  search_service.facet_suggest_response(@facet.key, params[:query_fragment])
+                else
+                  @response = search_service.facet_field_response(@facet.key)
+                end
+    set_facet_data
+
+    respond_to do |format|
+      format.html { render layout: false }
+      format.json { render 'facet' }
+    end
+  end
+
```

Comment:
Do we need both this action and the existing `facet` action? Since this one already has a `if query_fragment.present?` branch, can't it serve to display results whether or not a facet search fragment was input?  And be combined with `#facet` method, to have only one method serving facet responses instead of two?

Reply:
Good point!  I added a fourth commit combining these two actions back into a single `facet` action.

Reply:
love it, thanks!